ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -109,26 +113,49 @@ public void generateType(final Ds3Type typeEntry) throws IOException {
         final Template typeTemplate = config.getTemplate("TypeEnumConstant.tmplt");
         final Type type = TypeConverter.toType(typeEntry);
 
-        final Path outputPath = getOutputPath(type);
+        final Path outputPath = getTypeOutputPath(type);
+
+        final OutputStream outStream = fileUtils.getOutputFile(outputPath);
+        final Writer writer = new OutputStreamWriter(outStream);
+        try {
+            typeTemplate.process(type, writer);
+        } catch (final NullPointerException e) {
+            LOG.error("Encountered NullPointerException while processing template " + typeTemplate.getName(), e);
+            e.printStackTrace();
```

Comment:
Since you are appending the exception in the logging statement, you do not have to then print the stack trace.  This will actually result in the stack being printing twice, depending on how the logger is configured.